### PR TITLE
#573: Add toApp, toResponse methods to status

### LINKED ISSUE #573: Add methods to `Status`
The following methods can be added to Status

- [x] `toResponse` which converts the `Status` to a `Response`
- [x] `toApp` which converts the `Status` to `HttpApp`

### PR DESCRIPTION
**Pull request by jgoday**
Fixes #573.
Added toApp (UHttpApp) and toResponse (UResponse) to zhttp.http.Status.

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/Status.scala [hunk 1]
```diff
@@ -3,6 +3,10 @@ package zhttp.http
 import io.netty.handler.codec.http.HttpResponseStatus
 
 sealed trait Status extends Product with Serializable { self =>
+  def toApp: UHttpApp = HttpApp.status(self)
+
+  def toResponse: UResponse = Response(self)
+
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Status.scala:16]: a bit of scala docs would be great!

File: zio-http/src/test/scala/zhttp/http/StatusSpec.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  def toResponseSpec =
+    suite("toResponse")(
+      test("ok")(assert(Status.OK.toResponse.status)(equalTo(Status.OK))),
+    )
+
+  def toAppSpec = {
+    suite("toApp")(
+      testM("ok") {
+        val res = Status.OK.toApp.getResponse
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/http/StatusSpec.scala]:
```suggestion
        val res = Status.OK.toApp(Request())
```

File: zio-http/src/test/scala/zhttp/http/StatusSpec.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        assertM(res)(isResponse(responseStatus(200)))
+      },
+    )
+  }
+}
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/http/StatusSpec.scala:34]: would be nice to have the test use `Gen` for generating all kinds of statuses.

File: zio-http/src/test/scala/zhttp/http/StatusSpec.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package zhttp.http
+
+import zhttp.internal.HttpMessageAssertions
+import zhttp.service.EventLoopGroup
+import zio.test.Assertion._
+import zio.test._
+
+object StatusSpec extends DefaultRunnableSpec with HttpMessageAssertions {
+  private val env = EventLoopGroup.auto(1)
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/http/StatusSpec.scala]: No need to use a real server for these tests. An HttpApp can simply be called like a function.

```scala
val app : HttpApp[_, _] = ???
app(Request()) // ZIO[_, _, Response]
```
Something like above.

File: zio-http/src/main/scala/zhttp/http/Status.scala (hunk 1 above)
Comment (jgoday) [reply zio-http/src/main/scala/zhttp/http/Status.scala:16]: Sure, added some brief doc in the last commit

File: zio-http/src/test/scala/zhttp/http/StatusSpec.scala (hunk 3 above)
Comment (jgoday) [reply zio-http/src/test/scala/zhttp/http/StatusSpec.scala:34]: Ok, changed to use HttpGen.status

File: zio-http/src/test/scala/zhttp/http/StatusSpec.scala (hunk 4 above)
Comment (jgoday) [reply zio-http/src/test/scala/zhttp/http/StatusSpec.scala]: Oh, that is really good to know!

**Review by tusharmath — APPROVED** [review]

**Pull request merged**